File: engine/model/src/main/scala/tools/packet/message/NotifyMarriage.scala
package tools.packet.message

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class NotifyMarriage(private var _type: Int, private var _characterName: String) extends PacketInput {
  def theType: Int = _type

  def characterName: String = _characterName

  override def opcode(): SendOpcode = SendOpcode.NOTIFY_MARRIAGE
}

File: engine/engine-model/src/main/java/tools/packet/message/NotifyMarriage.java
package tools.packet.message;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record NotifyMarriage(Integer theType, String characterName) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NOTIFY_MARRIAGE;
   }
}